src/main/webapp/scripts/analyze.js

```diff
@@ -0,0 +1,44 @@
+function analyzeVideo() {
+  const videoId = document.getElementById('video-url').value.split('v=')[1];
+  const baseUrl = window.location.origin;
+  let url = new URL('/sentiment', baseUrl);
+  url.searchParams.append('video-id', videoId);
+
+  fetch(url)
+  .then(response => {
+    if (response.status >= 200 && response.status <= 299) {
+      return response.json();
+    } else if (response.status >= 400 && response.status <= 499) {
+      throw InvalidVideoException(response.statusText);
+    } else {
+      throw ServletFailureException(response.statusText);
```

Comment:
Whoa didn't know fetch does not catch all HTTP errors. The http responses in the backend are descriptive so I believe it is ok to just do a generic JS error and log it. No need for the custom errors.
Found in https://www.tjvantoll.com/2015/09/13/fetch-and-errors/
```
function handleErrors(response) {
    if (!response.ok) {
        throw Error(response.statusText);
    }
    return response;
}
fetch("http://httpstat.us/500")
    .then(handleErrors)
    .then(response => console.log("ok") )
    .catch(error => console.log(error) );
```

Reply:
After this you can clean up your test from multiple types of http errors to just one.